staging/src/k8s.io/client-go/rest/with_retry.go

```diff
@@ -278,20 +282,31 @@ func getRetryReason(retries, seconds int, resp *http.Response, err error) string
 	}
 }
 
-func readAndCloseResponseBody(resp *http.Response) {
-	if resp == nil {
-		return
-	}
+type responseCloser struct {
+	once sync.Once
```

Comment:
I think that we can close it multiple times, we are not doing this in parallel, no?